jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -663,7 +663,15 @@ public void run()
                 failure = x;
             }
             if (failure != null)
-                request._callback.failed(failure);
+            {
+                try (AutoLock ignored = _lock.lock())
+                {
+                    if (request._httpChannel != null && !request._httpChannel._completed)
+                        request._callback.failed(failure);
+                    else if (LOG.isDebugEnabled())
+                        LOG.debug("Process failed", failure);
+                }
+            }
```

Comment:
Potentially it would be neater to put this logic into a method:
```suggestion
            request._callback.ensureCompleted(failure);
```